dbt/adapters/bigquery/relation_configs/_options.py

```diff
@@ -158,4 +168,4 @@ class BigQueryOptionsConfigChange(RelationConfigChange):
 
     @property
     def requires_full_refresh(self) -> bool:
-        return False
+        return self.action != RelationConfigChangeAction.alter
```

Comment:
I would swap this one too, e.g. `self.action == RelationConfigChangeAction.create`.